src/pollyCLI/pollyintegration.cpp

```diff
@@ -0,0 +1,362 @@
+#include "pollyintegration.h"
+
+PollyIntegration::PollyIntegration(): nodePath(""), jsPath("")
+{
+    credFile = QStandardPaths::writableLocation(QStandardPaths::QStandardPaths::GenericConfigLocation) + QDir::separator() + "cred_file";
+
+    nodePath = QStandardPaths::findExecutable("node");
+    jsPath = qApp->applicationDirPath() + QDir::separator() + "index.js";
+
+    #ifdef Q_OS_WIN
+      if(!QStandardPaths::findExecutable("node", QStringList() << qApp->applicationDirPath()).isEmpty())
+        nodePath = qApp->applicationDirPath() + QDir::separator() + "node.exe";
+    #endif
+
+    #ifdef Q_OS_LINUX
+      if(!QStandardPaths::findExecutable("node", QStringList() << qApp->applicationDirPath()).isEmpty())
+          nodePath = qApp->applicationDirPath() + QDir::separator() + "node";
+    #endif
+
+    #ifdef Q_OS_MAC
+      QString binDir = qApp->applicationDirPath() + QDir::separator() + ".." + QDir::separator() + ".." + QDir::separator() + ".." + QDir::separator();
+      if(!QStandardPaths::findExecutable("node", QStringList() << binDir + "node_bin" + QDir::separator() ).isEmpty())
+        nodePath = binDir + "node_bin" + QDir::separator() + "node";
+
+      jsPath = binDir  + "index.js";
+    #endif
+}
+
+PollyIntegration::~PollyIntegration()
+{
+    qDebug()<<"exiting PollyIntegration now....";
+}
+
+// name OF FUNCTION: run_qt_process
+// PURPOSE:
+//    This function uses Qprocess from Qt library to execute terminal commands from C++ and
+// Returns its output as QByteArray
+// CALLS TO: none
+//
+// CALLED FROM: multiple functions in this script
+
+QByteArray PollyIntegration::run_qt_process(QString command, QStringList args){
+
+    // e.g: command = "authenticate", "get_Project_names" etc
+    // e.g: args = username, password, projectName  etc
+    QProcess process;
+    QStringList arg;
+    arg << jsPath; // where index.js files is placed
+    arg << command; // what command to pass to index.js. eg. authenticate
+    arg << args; // required params by that command . eg username, password
+
+    // nodePath = "PATH_OF_MAVEN/bin/node.exe"
+    process.setProgram(nodePath);
+    process.setArguments(arg);
+
+    qDebug () << "program: " <<  process.program()  << "args: " << process.arguments();
+
+    process.start();
+    process.waitForFinished(-1);
+    QByteArray result = process.readAllStandardOutput();
+    QByteArray result2 = process.readAllStandardError();
+
+    qDebug() << "output: " << result << endl;
+    qDebug()<< "error: " << result2 << endl;
+    return result;
+}
+
+// name OF FUNCTION: get_run_id
+// PURPOSE:
+//    This function parses the output of "createproject" command returned by run_qt_process
+// this command is used to create a new project on polly
+// this function Returns project id for the new project..
+// CALLS TO: none
+//
+// CALLED FROM: createProjectOnPolly
+
+
+QString PollyIntegration::get_run_id(QByteArray result){
```

Comment:
@Ky5595 pass **result** by **const** reference becauze size of **result** can be larger which will unnecessarily be copied every time.